src/message-card/message-card.tsx

```diff
@@ -0,0 +1,144 @@
+  StyledParagraphContainer,
+} from './styled-components';
+import { Button as DefaultButton, SHAPE, SIZE } from '../button';
+import { useStyletron } from '../styles/index.js';
+import { ThemeProvider, LightTheme } from '../';
+import { getBackgroundColorType } from './utils';
+import { colors } from '../tokens';
+import { getOverrides } from '../helpers/overrides';
+import { IMAGE_LAYOUT, BACKGROUND_COLOR_TYPE, BUTTON_KIND } from './constants';
+import type { MessageCardProps } from './types';
+
+const ButtonAlwaysLightTheme = ({ children, ...restProps }) => (
+  <ThemeProvider theme={LightTheme}>
+    <DefaultButton {...restProps}>{children}</DefaultButton>
+  </ThemeProvider>
+);
+
+const MessageCard = ({
+  backgroundColor = colors.white,
+  backgroundColorType: backgroundColorTypeProp,
+  buttonKind = BUTTON_KIND.secondary,
+  buttonLabel,
+  heading,
+  image,
+  onClick,
+  overrides = {},
+  paragraph,
+}: MessageCardProps) => {
+  const { src, layout = IMAGE_LAYOUT.top, backgroundPosition } = image || {};
+
+  const [Root, RootProps] = getOverrides(overrides.Root, StyledRoot);
+  const [ContentContainer, ContentContainerProps] = getOverrides(
+    overrides.ContentContainer,
+    StyledContentContainer
+  );
+  const [HeadingContainer, HeadingContainerProps] = getOverrides(
+    overrides.HeadingContainer,
+    StyledHeadingContainer
+  );
+  const [ParagraphContainer, ParagraphContainerProps] = getOverrides(
+    overrides.ParagraphContainer,
+    StyledParagraphContainer
+  );
+  const [Image, ImageProps] = getOverrides(overrides.Image, StyledImage);
+  const [Button, ButtonProps] = getOverrides(overrides.Button, ButtonAlwaysLightTheme);
+
+  const [, theme] = useStyletron();
+
+  let backgroundColorType = backgroundColorTypeProp || getBackgroundColorType(backgroundColor);
+  if (!backgroundColorType) {
+    backgroundColorType = BACKGROUND_COLOR_TYPE.dark;
+    if (__DEV__) {
+      console.warn(
+        `The provided value for 'backgroundColor', ${backgroundColor}, is not recognized as a \
+        Base Web primitive color. Please use the 'backgroundColorType' prop to indicate whether\
+         this color is light or dark so the rest of the component can be styled accordingly.`
+      );
+    }
+  }
+
+  let buttonColors;
+  if (buttonKind === BUTTON_KIND.tertiary && backgroundColorType === BACKGROUND_COLOR_TYPE.dark) {
+    buttonColors = {
+      color: theme.colors.contentOnColor,
+      backgroundColor: theme.colors.buttonTertiaryFill,
+    };
+  }
+  if (buttonKind !== BUTTON_KIND.tertiary && backgroundColor !== colors.white) {
+    buttonColors = {
+      color: theme.colors.contentOnColorInverse,
+      backgroundColor: theme.colors.backgroundAlwaysLight,
+    };
+  }
+
+  return (
+    <Root
+      onClick={onClick}
+      $backgroundColor={backgroundColor}
+      $backgroundColorType={backgroundColorType}
+      $imageLayout={layout}
+      {...RootProps}
+    >
+      {image && (
+        <Image
+          $src={src}
```

Comment:
I believe a `role="img"` and `aria-label` would be good https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role

Reply:
Yea I think you're right. I wasn't eager to add another required prop to the component, but the image really is more than just "background" decoration. I'll add this in.